sdk/core/azure-core/azure/core/utils/_pipeline_transport_rest_shared_async.py

```diff
@@ -0,0 +1,64 @@
+# coding=utf-8
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import asyncio
+from collections.abc import AsyncIterator
+from typing import TYPE_CHECKING, Any
+from ..pipeline import PipelineContext, PipelineRequest, PipelineResponse
+from ..pipeline._tools_async import await_result as _await_result
+
+if TYPE_CHECKING:
+    from typing import List
+    from ..pipeline._tools_async import HTTPResponseType
+    from ..pipeline.policies import SansIOHTTPPolicy
+
+class _PartGenerator(AsyncIterator):
+    """Until parts is a real async iterator, wrap the sync call.
+
+    :param parts: An iterable of parts
+    """
+
+    def __init__(self, response: "HTTPResponseType", default_http_response_type: Any) -> None:
```

Comment:
Does default_http_response_type have a default value?

Reply:
I don't have a default value in the shared code, since the default value is different between old and new responses. However, whenever I call the async PartGeneratr ([old](https://github.com/Azure/azure-sdk-for-python/blob/e26735c832f9949de6fdd0a8d8c02bc5c8f682e2/sdk/core/azure-core/azure/core/pipeline/transport/_base_async.py#L99) and [new](https://github.com/Azure/azure-sdk-for-python/blob/e26735c832f9949de6fdd0a8d8c02bc5c8f682e2/sdk/core/azure-core/azure/core/rest/_http_response_impl_async.py#L54)), I pass in the default